ts/components/AddServerDialog.tsx

```diff
@@ -0,0 +1,273 @@
+        if (!this.state.success) {
+          this.showView('default');
+
+          this.setState({
+            connecting: false,
+            success: false,
+            error: i18n('connectToServerFail'),
+          });
+
+          return;
+        }
+      }, max_connection_duration);
+
+      connectionResult
+        .then(() => {
+          clearTimeout(connectionTimeout);
+
+          if (this.state.connecting) {
+            this.setState({
+              success: true,
+            });
+            window.pushToast({
+              title: i18n('connectToServerSuccess'),
+              id: 'connectToServerSuccess',
+              type: 'success',
+            });
+            this.closeDialog();
+          }
+        })
+        .catch(error => {
+          clearTimeout(connectionTimeout);
+
+          this.showView('default');
+          this.setState({
+            connecting: false,
+            success: false,
+            error: error,
+          });
+        });
+    }
+  }
+
+  private showError() {
+    const message = this.state.error;
+    return (
+      <>
+        {message ? (
+          <>
+            <div className="session-label danger">{message}</div>
+            <div className="spacer-lg" />
+          </>
+        ) : null}
+      </>
+    );
+    // if (_.isEmpty(message)) {
+    //   this.$('.error').text('');
+    //   this.$('.error').hide();
+    // } else {
+    //   this.$('.error').text(`Error: ${message}`);
+    //   this.$('.error').show();
+    // }
+    // $('input').focus();
+  }
+
+  private onKeyUp(event: any) {
+    switch (event.key) {
+      case 'Enter':
+        if (this.state.view == 'default') {
+          this.showView('connecting');
+        }
+        break;
+      case 'Esc':
+      case 'Escape':
+        this.closeDialog();
+        break;
+      default:
+    }
+  }
+
+  private closeDialog() {
+    window.removeEventListener('keyup', this.onKeyUp);
+    this.props.onClose();
+  }
+
+  private async attemptConnection(serverUrl: string, channelId: number) {
+    const { i18n } = this.props;
+
+    const rawServerUrl = serverUrl
+      .replace(/^https?:\/\//i, '')
+      .replace(/[/\\]+$/i, '');
+    const sslServerUrl = `https://${rawServerUrl}`;
+    const conversationId = `publicChat:${channelId}@${rawServerUrl}`;
+
+    const conversationExists = window.ConversationController.get(
+      conversationId
+    );
+    if (conversationExists) {
+      // We are already a member of this public chat
+      return new Promise((resolve, reject) => {
+        if (false) {
+          resolve();
+        }
+        reject(i18n('publicChatExists'));
+      });
+    }
+
+    const serverAPI = await window.lokiPublicChatAPI.findOrCreateServer(
+      sslServerUrl
+    );
+    if (!serverAPI) {
+      // Url incorrect or server not compatible
+      return new Promise((resolve, reject) => {
+        if (false) {
+          resolve();
+        }
+        reject(i18n('connectToServerFail'));
```

Comment:
don't see the use of the resolve() here

Reply:
tslint forces resolve and reject when creating a promise

Reply:
Fixed